pages/hub/apps/guides/advanced/live-tutorial/data-retrieval.mdx

```diff
@@ -0,0 +1,313 @@
+import { Callout } from 'components';
+import { Tabs } from 'nextra-theme-docs'
+
+# Adding Live. Step 2: Live Data Retrieval
+
+Here, we'll explain how to display live games and their markets.
+You can find more undercover details in the related documentation: [Live Data Retrieval](../live/data-retrieval).
+
+The data structure is nearly identical to the one used in the main Subgraph for prematch core contracts.
+
+<Callout type="info">
+  Key differences in the entities structure comparing to prematch subgraphs:
+
+  - Removed all entities and events related to bets (bets, freebets, expresses) and contracts (cores, liquidity pools).
+  - Removed `selection` entity and its relations.
+  - Entity `condition` doesn't have an `isExpressForbidden` field.
+  - Event `GameShifted` changed to `GameUpdated` (triggers by changing game status and/or start time).
+  - Event `ConditionShifted` completely removed.
+</Callout>
+
+## Load live games
+
+If you've utilized fields excluded from the Live subgraph, we recommend splitting data fetching into `games` and `conditions`.
+This allows you to reuse game/sport queries for both Prematch and Live requests.
+
+Here's an example of a universal hook to retrieve events from both Prematch and Live feeds:
+
+<Tabs items={[ 'useSportsAndEvents.ts', 'query.ts' ]}>
+<Tabs.Tab>
+```ts {6,11,17-19,22-23,29} copy
+import { universalGamesQuery } from './query'
+
+const useSportsAndEvents = (props) => {
```

Comment:
why is it sports and events if you get only games?

Reply:
https://github.com/Azuro-protocol/gem-docs/pull/155/commits/f9e8ff61f1172594f6046f4b2fbbfab8f0422e1f